sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/AvroSerializer.java

```diff
@@ -7,13 +7,32 @@
 import com.azure.core.util.serializer.TypeReference;
 import reactor.core.publisher.Mono;
 
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.InputStream;
 import java.io.OutputStream;
 
 /**
  * Generic interface covering basic Avro serialization and deserialization methods.
  */
 public interface AvroSerializer extends ObjectSerializer {
+    /**
+     * Reads an Avro byte array into its object representation.
+     *
+     * @param data Avro byte array.
+     * @param typeReference {@link TypeReference} representing the object.
+     * @param <T> Type of the object.
+     * @return The object represented by the deserialized Avro byte array.
+     */
+    @Override
+    default <T> T deserialize(byte[] data, TypeReference<T> typeReference) {
+        if (data == null) {
+            return null;
+        }
+
+        return deserialize(new ByteArrayInputStream(data), typeReference);
+    }
```

Comment:
Since the same default implementation is in `ObjectSerializer`, is there a need to override here again?